### PR TITLE
Add Promo for Desktop

### PR DESCRIPTION
**Pull request by mkaz**
Create a promo spot for the Desktop app, initial display is within the Editor

### CODE REVIEW

**Comment by jasmussen** [comment]
Added a small visual fix. Now looks good to me from a design point of view! 

screen shot 2016-01-19 at 18 24 05

File: client/components/app-promo/index.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		store.set('desktop_promo_dismissed', true);
+	},
+
+	render() {
+		if ( this.state.dismissed ) {
+			return null;
+		}
+
+		var classes = classNames( 'app-promo', {
+			'dismissed': false
```
Comment (timmyc) [inline client/components/app-promo/index.jsx]: if `dismissed` is always false, then is `classNames` and the `classes` variable even needed here?  As is `classes` will always be `app-promo`.

File: client/components/app-promo/index.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import store from 'store';
+
+export default React.createClass( {
+
+	displayName: 'AppPromo',
+
+	getInitialState: function() {
+		var has_dismissed = store.get('desktop_promo_dismissed');
+		var promo_options = [
+			{ promo_code: 'a0001', message: 'WordPress.com your way  — apps now available in three delicious flavors — desktop, iOS, and Android.' },
```
Comment (timmyc) [inline client/components/app-promo/index.jsx]: Do these messages need to be wrapped in a `this.translate` call?

File: client/components/app-promo/index.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import noop from 'lodash/utility/noop';
+import Gridicon from 'components/gridicon';
+
+import analytics from 'analytics';
+import store from 'store';
+
+export default React.createClass( {
+
+	displayName: 'AppPromo',
+
```
Comment (timmyc) [inline client/components/app-promo/index.jsx]: Looks like there is a prop of `location` being passed in, there should be a `propTypes` declaration up here, and `location` might be set to `isRequired` to properly record the track event.

File: client/components/app-promo/index.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import assign from 'lodash/object/assign';
```
Comment (timmyc) [inline client/components/app-promo/index.jsx]: Looks like this dependency is not being used and could be removed.

File: client/components/app-promo/index.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import classNames from 'classnames';
+import noop from 'lodash/utility/noop';
```
Comment (timmyc) [inline client/components/app-promo/index.jsx]: Likewise, `noop` is not being used and can be removed.

**Comment by timmyc** [comment]
Took the component for a test drive and works well in my testing.  Left a few notes for some revisions on the code too.

Some other observations - the component uses ES6 structure :tada: - but is still using `var` declarations which could be switched over to `let` and `const`.  And the function syntax could also be further ES6-ified like `getInitialState: function() {` can become `getInitialState() {` 

Also I was wondering if this notice should also be shown in the context of the Desktop App or not?  I suppose it could still be promoting the other native apps, but wasn't sure if it should be feature flagged on desktop or not.

And we have also used localStorage to store dismiss state like this in various Stats features, but obviously the dismiss isn't truly sticky if localStore ever gets purged, or obviously doesn't work across devices/browsers.  There is the [Preferences](https://github.com/Automattic/wp-calypso/tree/master/client/lib/preferences) library which allows persisting random flags like this to the API - which if we adopt these promos further throughout Calypso it might be nice to back the dismiss state in the API.

File: client/components/app-promo/index.jsx (hunk 3 above)
Comment (mkaz) [inline client/components/app-promo/index.jsx]: Added propTypes / isRequired.  Thanks for the review!

File: client/components/app-promo/index.jsx (hunk 5 above)
Comment (mkaz) [inline client/components/app-promo/index.jsx]: Removed.

File: client/components/app-promo/index.jsx (hunk 4 above)
Comment (mkaz) [inline client/components/app-promo/index.jsx]: Removed.

**Comment by kellychoffman** [comment]
> Get WordPress.com apps for desktop and mobile: download now!
> WordPress.com apps now available for desktop and mobile: download now!

For these two: I think it would make it less spammy sounding if we removed the 'Download Now!'

Also, perhaps it could use some more unified padding:

screen shot 2016-01-19 at 1 05 59 pm

**Comment by mkaz** [comment]
Thanks for the review @timmyc 

Using localStorage was used since this is our first test of the promo spot. I'm not sure if we want to persist longer across other devices, yet. Ads are more effective when seen a few times, so switching device/browser and seeing it again is not necessarily bad.

**Comment by jasmussen** [comment]
Improved the padding a bit, good call @kellychoffman thanks:

screen shot 2016-01-19 at 20 59 35

**Comment by kellychoffman** [comment]
:+1: Looks good, @jasmussen

File: client/components/app-promo/index.jsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	},
+
+	getInitialState: function() {
+		var show_promo = true;
+
+		if ( store.get( 'desktop_promo_dismissed' ) ) {
+			show_promo = false;
+		}
+
+		if ( userUtils.getLocaleSlug() !== 'en' ) {
```
Comment (mjangda) [inline client/components/app-promo/index.jsx]: Minor optimization: we could combine these three checks into `if/else if` blocks.

**Comment by mjangda** [comment]
One minor optimization comment added. Tested and works well.

Otherwise :shipit:

**Pull request merged**